scripts/dotnet-cli-build/CompileTargets.cs

```diff
@@ -60,16 +62,18 @@ public static BuildTargetResult CompileCoreHost(BuildTargetContext c)
             Mkdirp(cmakeOut);
 
             var configuration = c.BuildContext.Get<string>("Configuration");
+            var architecture = PlatformServices.Default.Runtime.RuntimeArchitecture;
 
             // Run the build
             if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
             {
                 // Why does Windows directly call cmake but Linux/Mac calls "build.sh" in the corehost dir?
                 // See the comment in "src/corehost/build.sh" for details. It doesn't work for some reason.
+                var visualStudio = IsWinx86 ? "Visual Studio 14 2015" : "Visual Studio 14 2015 Win64";
                 ExecIn(cmakeOut, "cmake",
                     Path.Combine(c.BuildContext.BuildDirectory, "src", "corehost"),
                     "-G",
-                    "Visual Studio 14 2015 Win64");
+                    visualStudio);
 
```

Comment:
Minor NIT: If it is not too hard, could you please add this, while you are incorporating other feedback changes, thanks! I don't want to do a PR just for this:

-DCLI_CMAKE_PLATFORM_ARCH_I386=1
-DCLI_CMAKE_PLATFORM_ARCH_AMD64=1

Example: https://github.com/dotnet/coreclr/blob/master/CMakeLists.txt#L546